### PR TITLE
Fix JWT authentication when loading exercise templates

### LINKED ISSUE #37: Fix the loading of exercise templates now that they require JWT auth (in the gitmanager backend)
Radar needs to load the exercise template (base code, skeleton code for the students in assignments) from the Git manager (since January 2022, A+ v1.12). Git manager requires JWT authentication for it. Implement the token in Radar.

The token was started in this commit:
https://github.com/apluslms/radar/commit/bbda3f1d7c392893fd4b57fa9059418c2c79a642#diff-86a6d252027130bb3e9470141551defce99eb3a84d38ec135079baa05d42677d
But the permissions were forgotten and the token does not yet work in production environments. Add the course instance READ permissions to the token. This requires the A+ course instance ID as a parameter.

I have started this fix, but I haven't finished it:
https://github.com/Mankro/radar/commit/7c89b49212645726390402af561628a3ce547969

https://github.com/apluslms/radar/blob/5f911a3b199c436cbc46bbf2c95e02584e640fb6/provider/aplus.py#L82

```python
        permissions = Permissions()
        permissions.instances.add(Permission.READ, id=1) #TODO A+ course instance id
        response = aplus_get(url, permissions=permissions, timeout=(4, 10))
```

### PR DESCRIPTION
**Pull request by PasiSa**
The permission attributes were missing from JWT token sent to Gitmanager. Fixes #37.

# Description

**What?**

As discussed in the issue description. Will do the needed aplus-auth config changes to run-radar docker setup separately.

**Why?**

[ANSWER HERE]

**How?**

[ANSWER HERE]

Fixes #<issue number>

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [ ] Manual testing.

[ADD A DESCRIPTION ABOUT WHAT YOU TESTED MANUALLY]

**Did you test the changes in**

- [ ] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [ ] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

File: provider/aplus.py [hunk 1]
```diff
@@ -105,7 +108,7 @@ def request_template_content(url):
 # "|en:<external url to template in english>|fi:<external url to template in finnish>|"
 # Therefore, if i18n exercise template urls are found,
 # we use a heuristic to first select the english template and if not available, the finnish template, and if not available, fail miserably
-def load_exercise_template_from_api_data(exercise_data):
+def load_exercise_template_from_api_data(exercise_data, course):
```
Comment (Mankro) [inline provider/aplus.py:111]: Below, in the function `load_exercise_template()`, this function `load_exercise_template_from_api_data()` is called, but you have not updated the parameters in the call. Doesn't that cause a crash in `load_exercise_template()`? Please check this.

I guess `load_exercise_template()` already has easy access to the `course` object via `exercise.course` there.

File: provider/aplus.py [hunk 2]
```diff
@@ -159,7 +162,7 @@ def load_exercise_template(exercise, config, invalidate_cache=False):
     return template_source
 
 
-def get_radar_config(exercise_data):
+def get_radar_config(exercise_data, course):
```
Comment (Mankro) [inline provider/aplus.py:165]: In the function `provider.tasks.create_submission()`, this function `get_radar_config()` is called, but you have not updated the parameters in the call. The function already contains a `course` variable so adding it to the function call should be easy.

**Review by Mankro — CHANGES_REQUESTED** [review]
This looks great! I just think you forgot two function calls when you modified the parameters in the function definitions. Python crashes when the function call does not include all mandatory positional parameters.

**Comment by PasiSa** [comment]
> This looks great! I just think you forgot two function calls when you modified the parameters in the function definitions. Python crashes when the function call does not include all mandatory positional parameters.

Fixed as suggested (and tested too, believe it or not :-)

**Pull request merged**